### PR TITLE
Rust runtime tidy-ups

### PR DESCRIPTION
**Pull request by daviddrysdale**
As independent commits so things can be easily dropped if needed.

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/server/rust/oak_runtime/src/node/wasm/mod.rs [hunk 1]
```diff
@@ -478,31 +478,39 @@ impl WasmInterface {
...
+        for (i, handle) in handles.into_iter().enumerate() {
+            if let Some(channel) = self.readers.get(&handle).copied() {
+                valid_pos.push(i);
+                valid_channels.push(channel);
+            }
+        }
+        let valid_statuses = self.runtime.wait_on_channels(&valid_channels)?;
+        for i in 0..valid_channels.len() {
+            all_statuses[valid_pos[i]] = valid_statuses[i];
+        }
```
Comment (tiziano88) [inline oak/server/rust/oak_runtime/src/node/wasm/mod.rs:502]: Incidentally, I think when https://github.com/project-oak/oak/issues/803 is fixed, this should become simpler. cc @rbehjati

File: oak/server/rust/oak_runtime/src/node/wasm/mod.rs [hunk 2]
```diff
@@ -71,9 +70,9 @@ struct WasmInterface {
     pretty_name: String,
 
     /// Reader channel mappings to unique u64 handles
-    readers: HashMap<AbiHandle, Handle>,
+    readers: HashMap<oak_abi::Handle, Handle>,
```
Comment (tiziano88) [inline oak/server/rust/oak_runtime/src/node/wasm/mod.rs:73]: IIRC you were planning to clean up (remove?) this mapping at some point, right?

Comment (daviddrysdale) [reply oak/server/rust/oak_runtime/src/node/wasm/mod.rs:73]: Yep, incoming once I find a decent block of time to concentrate in.

File: oak/server/rust/oak_runtime/src/node/wasm/mod.rs (hunk 1 above)
Comment (daviddrysdale) [reply oak/server/rust/oak_runtime/src/node/wasm/mod.rs:502]: SGTM.

**Pull request merged**